src/main/cc/wfa/virtual_people/core/model/utils/distributed_consistent_hashing_test.cc

```diff
@@ -0,0 +1,250 @@
+  // choice_id probability_before_normalized normalized_probability
+  // 0         0.8                           0.4
+  // 1         0.4                           0.2
+  // 2         0.4                           0.2
+  // 3         0.4                           0.2
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0.8);
+  distribution->emplace_back(1, 0.4);
+  distribution->emplace_back(2, 0.4);
+  distribution->emplace_back(3, 0.4);
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  absl::flat_hash_map<int32_t, int> output_counts = {
+      {0, 0},
+      {1, 0},
+      {2, 0},
+      {3, 0}};
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    int32_t output = hashing->Hash(std::to_string(seed));
+    // The output should always be one of [0, 1, 2, 3].
+    EXPECT_TRUE(output_counts.find(output) != output_counts.end());
+    ++output_counts[output];
+  }
+  // Absolute error more than 2% is very unlikely.
+  EXPECT_NEAR(
+      static_cast<double>(output_counts[0]) / static_cast<double>(kSeedNumber),
+      0.4, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(output_counts[1]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(output_counts[2]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+  EXPECT_NEAR(
+      static_cast<double>(output_counts[3]) / static_cast<double>(kSeedNumber),
+      0.2, 0.02);
+}
+
+TEST(DistributedConsistentHashingTest, TestZeroProbability) {
+  // Distribution:
+  // choice_id probability
+  // 0         0
+  // 1         1
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0);
+  distribution->emplace_back(1, 1);
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    EXPECT_EQ(hashing->Hash(std::to_string(seed)), 1);
+  }
+}
+
+TEST(DistributedConsistentHashingTest, TestZeroAfterNormalization) {
+  // Distribution:
+  // choice_id probability_before_normalized       normalized_probability
+  // 0         std::numeric_limits<double>::min()  0
+  // 1         std::numeric_limits<double>::max()  1
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, std::numeric_limits<double>::min());
+  distribution->emplace_back(1, std::numeric_limits<double>::max());
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    EXPECT_EQ(hashing->Hash(std::to_string(seed)), 1);
+  }
+}
+
+TEST(DistributedConsistentHashingTest,
+     TestOutputDistribution_NonConsecutiveChoiceId) {
```

Comment:
Please don't use underscores in googletest test names. https://github.com/google/googletest/blob/master/docs/faq.md#why-should-test-suite-names-and-test-names-not-contain-underscore

Reply:
Done.